src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_AssignmentOperator.cs

```diff
@@ -301,11 +301,13 @@ private BoundExpression VisitAssignmentOperator(BoundAssignmentOperator node, bo
             if (setMethod is null)
             {
                 var autoProp = (SourcePropertySymbolBase)property.OriginalDefinition;
-                Debug.Assert(autoProp.IsAutoPropertyWithGetAccessor,
+                Debug.Assert(autoProp.IsAutoPropertyWithGetAccessor || autoProp.FieldKeywordBackingField is not null,
```

Comment:
>|| autoProp.FieldKeywordBackingField is not null, [](http://example.com/codeflow?start=68&length=49)

Why this change is necessary? We are increasing the scope of this PR, adding more things to be concerned about, by calling APIs that trigger accessor binding in many different places. From the proposed plan: "Let’s avoid force binding accessors as much as possible. I think we should be able to get to the state when only GetFieldsToEmit can trigger the binding." Is this change made to support assignment to a read-only property in constructor? Let's not support this scenario for now. We'll focus on that scenario later, probably when dealing with initializers, or even as a separate task.

Reply:
@AlekseyTs The proposed plan also mentioned:

> Let’s focus only on scenarios that do not deal with presence of semicolon-only accessors. I.e. let’s just get the scenarios when the only or both accessors have explicit body (whatever syntactic form that body might take) working in presence of the field keyword. From declaration to execution.

The "to execution" part requires binding per my understanding. The added test that asserts for an expected output at runtime requires this change.

Reply:
> The "to execution" part requires binding per my understanding. The added test that asserts for an expected output at runtime requires this change.

Please be specific. What is the scenario that you think cannot work without this change?

Reply:
@AlekseyTs 

https://github.com/dotnet/roslyn/blob/a87af0b6c09e1141894a58dc9f55d27274d2daf6/src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs#L840-L860

Reply:
> [https://github.com/dotnet/roslyn/blob/a87af0b6c09e1141894a58dc9f55d27274d2daf6/src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs#L840-L860](https://github.com/dotnet/roslyn/blob/a87af0b6c09e1141894a58dc9f55d27274d2daf6/src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs#L840-L860)

This looks exactly as the scenario I outlined at the beginning of this thread. Here is the quote: "Is this change made to support assignment to a read-only property in constructor? Let's not support this scenario for now. We'll focus on that scenario later, probably when dealing with initializers, or even as a separate task." I think we will be able to get that working without extra binding as well. That will require extra work elsewhere. If you would like, we can do that work in this PR. However, if we start biting more and more work, we might never finish this PR.

Reply:
Sorry I didn't notice this part:

> Is this change made to support assignment to a read-only property in constructor? Let's not support this scenario for now.

I'm going to revert and disable the new test for now.